sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala

```diff
@@ -56,13 +60,20 @@ trait GpuBaseLimitExec extends LimitExec with GpuExec {
 
         override def next(): ColumnarBatch = {
           val batch = cbIter.next()
-          val result = if (batch.numRows() > remainingLimit) {
-            sliceBatch(batch)
-          } else {
-            batch
+          withResource(new NvtxWithMetrics("limit", NvtxColor.ORANGE, totalTime)) { _ =>
+            val result = if (batch.numRows() > remainingLimit) {
+              System.err.println(s"SLICE $remainingLimit/${batch.numRows()}")
+              sliceBatch(batch)
+            } else {
+              System.err.println(s"PASS $remainingLimit/${batch.numRows()}")
+              batch
+            }
+            numOutputBatches += 1
+            numOutputRows += result.numRows()
+            remainingLimit -= result.numRows()
+            System.err.println(s"RESULT $remainingLimit/${result.numRows()}")
```

Comment:
debug print

Reply:
sorry about that, I thought I had an off by one error when I tested the metrics with show.  Turns out it adds one to the limit in inserts.  I don't know why.  But `collect(1)` actually sets the limit to 2.  Odd